arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -306,12 +289,12 @@ public class OptionRaise(private val raise: Raise<None>) : Raise<None> by raise
  */
 public class IorRaise<Error> @PublishedApi internal constructor(
   @PublishedApi internal val combineError: (Error, Error) -> Error,
-  private val state: Atomic<Option<Error>>,
+  @PublishedApi internal val state: Atomic<Any?>,
```

Comment:
Would it be possible to keep this private?

Reply:
That's an interesting question... There is a subtle difference in behaviour if we keep this private (and thus implement `recover` as `ior(combineError, block).bind()`. Namely, if you `throw` inside a `recover`, the values won't get accumulated. Not sure which behaviour is more "correct", but it's important to note that the second behaviour, where throwing "rolls back" the accumulation is hard to replicate with contexts. Conceptually, `recover` should only replace the `Raise` instance you're dealing with, and nothing else, at least that's my understanding.